src/searcher/mod.rs

```diff
@@ -297,6 +334,60 @@ impl Searcher {
         MultiLine::new(self, query_context, slice, write_to).run()
     }
 
+    fn search_slice_callback(
+        &mut self,
+        query_context: QueryContext,
+        slice: &[u8],
+        callback: impl Fn(Node, &[u8], &Path),
+        path: &Path,
+    ) -> Result<(), ConfigError> {
+        self.check_config()?;
+
+        log::trace!("slice reader: searching via multiline strategy");
+        self.run_with_callback(query_context, slice, callback, path);
+
+        Ok(())
+    }
+
+    fn run_with_callback(
+        &self,
+        query_context: QueryContext,
+        slice: &[u8],
+        callback: impl Fn(Node, &[u8], &Path),
+        path: &Path,
+    ) {
+        let mut query_cursor = QueryCursor::new();
```

Comment:
This also feels very non-DRY wrt `MultiLine::run()` or wherever this stuff happens in "command-line mode"

I've slightly thought about encapsulating this before and seemed tricky wrt eg lifetimes